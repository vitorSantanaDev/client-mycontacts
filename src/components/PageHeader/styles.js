import styled, { css } from 'styled-components'

export const Container = styled.header`
	${({ theme }) => css`
		a {
			display: flex;
			align-items: center;
			text-decoration: none;
			cursor: pointer;

			span {
				font-weight: ${theme.font.bold};
				color: ${theme.colors.primary.main};
			}

			img {
				margin-right: 0.8rem;
				transform: rotate(-90deg);
			}
		}

		h1 {
			font-size: calc(${theme.font.sizes.xlarge} + 0.4rem);
		}
	`}
`
