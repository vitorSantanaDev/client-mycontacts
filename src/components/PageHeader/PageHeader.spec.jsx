import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderWithTheme from '../../utils/render-with-theme'

import PageHeader from '.'
import { screen } from '@testing-library/react'

describe('<Header />', () => {
	it('should render the Page Header component correctly', () => {
		renderWithTheme(
			<MemoryRouter>
				<PageHeader title="Novo contato" />
			</MemoryRouter>
		)

		expect(
			screen.getByRole('heading', { name: /Novo contato/i })
		).toBeInTheDocument()
	})
	it('should match snapshot', () => {
		const { container } = renderWithTheme(
			<MemoryRouter>
				<PageHeader title="Novo contato" />
			</MemoryRouter>
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
