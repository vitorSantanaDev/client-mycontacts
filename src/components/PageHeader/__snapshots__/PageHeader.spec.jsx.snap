// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c0 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0 a span {
  font-weight: 700;
  color: #5061FC;
}

.c0 a img {
  margin-right: 0.8rem;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c0 h1 {
  font-size: calc(2.0rem + 0.4rem);
}

<header
  class="c0"
>
  <a
    href="/"
  >
    <img
      alt="Back"
      src="arrow.svg"
    />
    <span>
      Voltar
    </span>
  </a>
  <h1>
    Novo contato
  </h1>
</header>
`;
